libs/net/models/Extensions/ConnectionModelExtensions.cs

```diff
@@ -82,12 +83,58 @@ public static string GetConfigurationValue(this API.Areas.Services.Models.DataLo
             {
                 JsonValueKind.String => (T)Convert.ChangeType($"{element.GetString()}".Trim(), typeof(T)),
                 JsonValueKind.Null or JsonValueKind.Undefined => default,
-                JsonValueKind.Number => element.GetInt64().ChangeType<T>(), // TODO: Handle different number types.
+                JsonValueKind.Number => ConvertNumberTo<T>(element),
                 JsonValueKind.True or JsonValueKind.False => (T)Convert.ChangeType($"{element.GetBoolean()}", typeof(T)),
                 _ => (T)Convert.ChangeType($"{element}", typeof(T)),
             };
         }
 
         throw new ConfigurationException($"Dictionary key '{key}' is not a valid JSON element");
     }
+
+    /// <summary>
+    /// Convert number to specified type.
+    /// This method handles enum values.
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    /// <param name="element"></param>
+    /// <returns></returns>
+    private static T? ConvertNumberTo<T>(JsonElement element)
```

Comment:
JSON doesn't understand different number types.  This method 'should' handle different number types better.